indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/RecordSupplierInputSource.java

```diff
@@ -68,16 +70,20 @@ public RecordSupplierInputSource(
     this.recordSupplier = recordSupplier;
     this.useEarliestOffset = useEarliestOffset;
     this.iteratorTimeoutMs = iteratorTimeoutMs;
+
     try {
       assignAndSeek(recordSupplier);
     }
     catch (InterruptedException e) {
-      throw new SamplerException(e, "Exception while seeking to partitions");
```

Comment:
It seems weird to throw a `SamplerException` from an `InputSource` implementation but apparently this impl is used only with the sampler. So I guess that is okay.

Reply:
Yeah, might have to revisit this at some point.

Reply:
Yeah, might have to revisit this at some point.